### PR TITLE
Canary deployments with flagger - core chart

### PR DESCRIPTION
**Pull request by AliakseiVenski**
## Description
Feature branch for canary deployments support with Flagger. New resources added to the chart chart:
- MetricTemplates

## Chart

Select the chart that you are modifying:
- [x] core
- [ ] dotnet-template
- [ ] pact-broker
- [ ] azure-functions

## Checklist
- [x] Description provided
- [ ] Linked issue
- [x] Chart version bumped
- [x] README.md updated with any new values or changes
- [ ] Updated template tests in `${CHART}/tests/*.py`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,5 +5,4 @@
 *.tgz
 *.lock
 **/*.lock
-.pytest_cache
```
Comment (RobinRojowiec) [inline .gitignore:8]: Can you please correct the ignore as in https://github.com/EcovadisCode/charts/pull/46/commits/a1b1ae4fa98379a8b1bff9dd00f6f9afc99f744b ?

File: charts/dotnet-core/Chart.yaml [hunk 2]
```diff
@@ -5,6 +5,6 @@ name: charts-dotnet-core
 type: application
 dependencies:
 - name: charts-core
-  version: 1.2.1
+  version: 2.0.0
```
Comment (Imroath) [inline charts/dotnet-core/Chart.yaml:8]: Update to dotnet-core chart needs to be in a separate PR.
That's the limitation of current pipeline. Since when it runs the charts-core chart is not yet published, dotnet-core chart can't reference it so the publish process will fail.

File: charts/core/templates/CRD/servicemonitor.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.global.servicemonitor.enabled -}}
+{{- if .Values.global.monitoring.servicemonitor.enabled -}}
```
Comment (Imroath) [inline charts/core/templates/CRD/servicemonitor.yaml:1]: Today at least lnm-engine and Traefik are are using ServiceMonitor. Can you please make sure that there is no regression in those due to this change?

**Review by Imroath — APPROVED** [review]
